fbpic/fields/cuda_methods.py

```diff
@@ -491,17 +491,17 @@ def cuda_compute_grad_a( a, grad_a_p, grad_a_m, grad_a_z, d_kr, d_kz, Nz, Nr ):
         grad_a_z[iz, ir] = 1j * a[iz, ir] * d_kz[iz, ir]
 
 @cuda.jit
-
 def cuda_convolve(chi_a, chi, a):
     # Cuda 2D grid
     iz, ir = cuda.grid(2)
 
     if (iz < chi.shape[0]) and (ir < chi.shape[1]):
         chi_a[iz, ir] += chi[iz, ir] * a[iz, ir]
 
-def cuda_copy_arrays(copy, array, Nz, Nr):
+@cuda.jit
+def cuda_copy_arrays(copy, array, Nz, Nr, m):
     # Cuda 2D grid
     iz, ir = cuda.grid(2)
 
     if (iz < Nz) and (ir < Nr) :
-        copy[iz, ir] = array[iz, ir]
+        copy[m, iz, ir] = array[iz, ir]
```

Comment:
This change with the index `m` is not strictly required here ; it's just a matter of personal preference.